examples/erc20/src/lib.rs

```diff
@@ -105,4 +105,12 @@ impl Erc20Example {
         self.pausable.when_not_paused()?;
         self.erc20.transfer_from(from, to, value).map_err(|e| e.into())
     }
+
+    fn supports_interface(
+        interface_id: FixedBytes<4>,
+    ) -> Result<bool, Vec<u8>> {
+        let interface_id = u32::from_be_bytes(*interface_id);
+        let supported = interface_id == <Erc20 as IErc20>::INTERFACE_ID;
```

Comment:
It also should check for IErc165 interface id

Reply:
I'm sure there are better ways to achieve the same, but something like this should work